syft/frameworks/torch/nn/functional.py

```diff
@@ -41,6 +41,50 @@ def dropout(input, p=0.5, training=True, inplace=False):
     return input
 
 
+def batch_norm(
+    input, running_mean, running_var, weight, bias, training, exponential_average_factor, eps
+):
+    """
+    Implementation of batch_norm
+
+    Note that exponential_average_factor is not supported for the moment and should be set to 0
+    when training a model. However for testing, this doesn't matter.
+    """
+    input = input.permute(1, 0, 2, 3)
+    input_shape = input.shape
+    input = input.reshape(input_shape[0], -1)
+    input = input.t()
+
+    if training:
+        assert exponential_average_factor == 0  # == momentum
+        mean = input.mean(dim=0)
+        var = input.var(dim=0) + eps
+    else:
+        mean = running_mean
+        var = running_var
+
+    x = None
+
+    C = 20
+
+    for i in range(80):
+        if x is not None:
+            y = C + 1 - var * (x * x)
+            x = y * x / C
+        else:
+            y = C + 1 - var
+            x = y / C
```

Comment:
Why not executing the else outside, and removing condition in the for loop, since it's 1 vs 79 times.

Reply:
I agree, however we have 80 for bad reason, in the near future we'll have 3 to 5 iterations :) 
I'm waiting for the FALCON team to do this :D